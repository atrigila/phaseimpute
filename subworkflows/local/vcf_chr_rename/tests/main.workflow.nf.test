nextflow_workflow {

    name "Test Workflow VCF_CHR_RENAME"
    script "../main.nf"
    workflow "VCF_CHR_RENAME"
    tag 'subworkflows'
    tag 'vcf_chr_rename'
    tag 'subworkflows/vcf_chr_rename'

    test("Should run without error") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: "input", genome:"GRCh37"],
                    "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/vcf/NA24385_sv.vcf.gz",
                    "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/vcf/NA24385_sv.vcf.gz.tbi"
                ])
                input[1] = file("assets/chr_rename.txt", exist: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.ch_region).match() }
            )
        }
    }
}
