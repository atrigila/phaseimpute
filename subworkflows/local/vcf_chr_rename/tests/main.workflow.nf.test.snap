{
    "Should run without error": {
        "content": null,
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-12T14:30:51.382997081"
    }
}